tests/std/tests/P1222R4_flat_set/test.cpp

```diff
@@ -166,6 +274,82 @@ void test_constructors() {
     assert_all_requirements_and_equals(flat_multiset<int>(std::move(b), allocator<int>{}), {-1, 1, 2, 7, 7, 7, 100});
 }
 
+template <iterator_pair_construction Choice>
+void test_allocator_extended_constructors() {
+    using vec = alternative_vector<int, allocator<int>, Choice>;
+    constexpr allocator<int> ator;
+    constexpr std::less<int> comp;
+    {
+        using fs = flat_set<int, std::less<int>, vec>;
+
+        fs s{3, 7, 1, 85, 222, 1};
+        fs s_expected{1, 3, 7, 85, 222};
+        vec v_raw{3, 7, 1, 85, 222, 1};
+        vec v_sorted_unique{1, 3, 7, 85, 222};
+
+        TEST_ASSERT(fs{comp, ator} == fs{});
+
+        TEST_ASSERT(fs{s, ator} == s_expected);
+        TEST_ASSERT(fs{s_expected, ator} == s_expected);
+        TEST_ASSERT(fs{std::move(s), ator} == s_expected);
+        TEST_ASSERT(fs{fs{s_expected}, ator} == s_expected);
+
+        TEST_ASSERT(fs{v_raw, ator} == s_expected);
+        TEST_ASSERT(fs{{3, 7, 1, 85, 222, 1}, ator} == s_expected);
+        TEST_ASSERT(fs{v_raw.begin(), v_raw.end(), ator} == s_expected);
+        TEST_ASSERT(fs{from_range, v_raw, ator} == s_expected);
+
+        TEST_ASSERT(fs{v_raw, comp, ator} == s_expected);
+        TEST_ASSERT(fs{{3, 7, 1, 85, 222, 1}, comp, ator} == s_expected);
+        TEST_ASSERT(fs{v_raw.begin(), v_raw.end(), comp, ator} == s_expected);
+        TEST_ASSERT(fs{from_range, v_raw, comp, ator} == s_expected);
+
+        TEST_ASSERT(fs{sorted_unique, v_sorted_unique, ator} == s_expected);
+        TEST_ASSERT(fs{sorted_unique, v_sorted_unique, comp, ator} == s_expected);
+        if constexpr (Choice == iterator_pair_construction::with_allocator) {
+            TEST_ASSERT(fs{sorted_unique, {1, 3, 7, 85, 222}, ator} == s_expected);
+            TEST_ASSERT(fs{sorted_unique, {1, 3, 7, 85, 222}, comp, ator} == s_expected);
+
+            TEST_ASSERT(fs{sorted_unique, v_sorted_unique.begin(), v_sorted_unique.end(), ator} == s_expected);
+            TEST_ASSERT(fs{sorted_unique, v_sorted_unique.begin(), v_sorted_unique.end(), comp, ator} == s_expected);
+        }
+    }
+    {
+        using fms = flat_multiset<int, std::less<int>, vec>;
+
+        fms s{3, 7, 1, 85, 222, 1};
+        fms s_expected{1, 1, 3, 7, 85, 222};
+        vec v_sorted_eq{1, 1, 3, 7, 85, 222};

```

Comment:
This is still missing the `v_raw` coverage that the `flat_set` tests above provide.